types/coin.go

```diff
@@ -100,6 +100,11 @@ func (coin Coin) Add(coinB Coin) Coin {
 	return Coin{coin.Denom, coin.Amount.Add(coinB.Amount)}
 }
 
+// AddAmount adds an amount to the Coin.
```

Comment:
what is the benefit to keeping this and the above function? Seems like we can do some cleanup here?

Reply:
`sdk.Coin.Add()` may remain for backward compatibility, and also needed for this situation:
```go
result := existingCoinA.Add(existingCoinB)
```

If there is no `sdk.Coin.Add()` and we only have `sdk.Coin.AddAmount()`, then it'd be:
```go
result := exisitingCoinA.AddAmount(existingCoinB.Amount)
```

This is error-prone since there is no assumption of `existingCoinA.Denom == existingCoinB.Denom`.

Both `sdk.Coin.Add()` and `sdk.Coin.AddAmount()` are suitable for different situations.

Reply:
`sdk.Coin.Add()` may remain for backward compatibility, and also needed for this situation:
```go
result := existingCoinA.Add(existingCoinB)
```

If there is no `sdk.Coin.Add()` and we only have `sdk.Coin.AddAmount()`, then it'd be:
```go
result := exisitingCoinA.AddAmount(existingCoinB.Amount)
```

This is error-prone since there is no assumption of `existingCoinA.Denom == existingCoinB.Denom`.

Both `sdk.Coin.Add()` and `sdk.Coin.AddAmount()` are suitable for different situations.